contrib/submit-protocol/src/App/MarketplaceForm.tsx

```diff
@@ -0,0 +1,190 @@
+  dropdown: {
+    width: 240,
+  },
+});
+
+export default class MarketplaceForm extends React.Component<IMarketplaceProps, IMarketplaceState> {
+  public state = {
+    uri: this.props.uri || "",
+    uriType: this.props.uriType as MarketplaceUriType || MarketplaceUriType.None,
+    protocolOptions: [
+      { key: "None", text: "None" },
+      { key: "jobsDivider", text: "-", itemType: DropdownMenuItemType.Divider },
+      { key: "jobsHeader", text: "Protocol Jobs", itemType: DropdownMenuItemType.Header },
+    ],
+    uriConfigCallout: false,
+  };
+
+  private uriConfigCalloutBtn = React.createRef<HTMLDivElement>();
+
+  public componentDidMount() {
+    this.setProtocolOptions(null as any);
+  }
+
+  public render() {
+    return (
+      <Fabric>
+        <Stack gap={5} horizontal={true} verticalAlign="center">
+          <Stack>
+            <Dropdown
+              className={styles.dropdown}
+              placeholder="Select a protocol config file"
+              options={this.state.protocolOptions}
+              onChange={this.selectProtocol}
+              disabled={this.props.disabled}
+            />
+          </Stack>
+          <Stack maxWidth="5px">
+            <div ref={this.uriConfigCalloutBtn}>
+              <DefaultButton
+                onClick={this.onConfigCallout}
+                iconProps={{iconName: "ConfigurationSolid"}}
+                text="URI"
+                disabled={this.props.disabled}
+              />
+            </div>
+            <Callout
+              role="alertdialog"
+              target={this.uriConfigCalloutBtn.current}
+              onDismiss={this.closeConfigCallout}
+              setInitialFocus={true}
+              hidden={!this.state.uriConfigCallout}
+            >
+              <Stack padding={20}>
+                <TextField
+                  label="Marketplace URI"
+                  prefix="https://api.github.com/repos/"
+                  description="Endpoint should be :owner/:repo/contents/:path?ref=:branch"
+                  onChange={this.setMarketplaceURI}
+                  onBlur={this.setProtocolOptions}
+                />
+              </Stack>
+            </Callout>
+          </Stack>
+        </Stack>
+      </Fabric>
+    );
+  }
+
+  private onConfigCallout = () => {
+    this.setState({uriConfigCallout: !this.state.uriConfigCallout});
+  }
+
+  private closeConfigCallout = () => {
+    this.setState({uriConfigCallout: false});
+  }
+
+  private setMarketplaceURI = (event: React.FormEvent<HTMLInputElement | HTMLTextAreaElement>, uri?: string) => {
+    if (uri !== undefined) {
+      this.setState({
+        uri: `https://api.github.com/repos/${uri}`,
+        uriType: MarketplaceUriType.GitHub,
+      });
+    }
+  }
+
+  private setProtocolOptions = async (event: React.FocusEvent<HTMLInputElement>) => {
+    const protocolList = await this.getProtocolList(this.state.uri, this.state.uriType);
+    if (protocolList) {
+      const protocolOptions = [
```

Comment:
```suggestion
      const protocolOptions: IDropdownOption[] = [
```

will diss the following any

Reply:
updated